OSDOCS#9497: editing canary rollout update page (#71004)
openshift/openshift-docs: modules/update-using-custom-machine-config-pools-about.adoc

[id="update-using-custom-machine-config-pools-about_{context}"]
= About performing a canary rollout update

This following steps outline the high-level workflow of the canary rollout update process:

. Create custom machine config pools (MCP) based on the worker pool.
+
[NOTE]
====
You can change the `maxUnavailable` setting in an MCP to specify the percentage or the number of machines that can be updating at any given time. The default is `1`.
====

. Add a node selector to the custom MCPs. For each node that you do not want to update simultaneously with the rest of the cluster, add a matching label to the nodes. This label associates the node to the MCP.
+
[IMPORTANT]
====
Do not remove the default worker label from the nodes. The nodes must have a role label to function properly in the cluster.
====

. Pause the MCPs you do not want to update as part of the update process.

. Perform the cluster update. The update process updates the MCPs that are not paused, including the control plane nodes.

. Test your applications on the updated nodes to ensure they are working as expected.

> . Unpause the remaining MCPs one at a time and test the applications on those nodes until all worker nodes are updated. Unpausing an MCP starts the update process for the nodes associated with that MCP.

Author: @JianLi-RH I can change it to something like the following:
```suggestion
. Unpause one of the remaining MCPs, wait for the nodes in that pool to finish updating, and test the applications on those nodes. 
Repeat this process until all worker nodes are updated.
```

Does that sound better?

> . Unpause the remaining MCPs one at a time and test the applications on those nodes until all worker nodes are updated. Unpausing an MCP starts the update process for the nodes associated with that MCP.

Reviewer: Very nice! Thank you!

> . Optionally, remove the custom label from updated nodes and delete the custom MCPs.

Author: If this is optional, should [this section](https://github.com/openshift/openshift-docs/pull/71004/files#diff-cc50790f38542369fded8e90bdb8dbbd70fe0e0ff6518577f270d53c06f7c245) be described as optional as well?

Reviewer 2: From Kathryn: 

Kathryn Alexander
:books:  [15 days ago](https://redhat-internal.slack.com/archives/C06J4CEB4EN/p1707942863354129?thread_ts=1707942460.929699&cid=C06J4CEB4EN)
For titles, I think we decided that we shouldn't use it. If modules are meant to be reusable, you can't always say that it's going to be optional.

There might be a PR open for our style guide that suggests against. I think @michaelryanpeter was going to handle that? Not sure if he has yet.

[id="update-using-custom-machine-config-pools-about_{context}"]
= About performing a canary rollout update

> This following steps outline the high-level workflow of the canary rollout update process:

Author: Good catch, oops
